File: src/main/java/com/ohgnarly/fileripper/models/FileDefinition.java
package com.ohgnarly.fileripper.models;

import com.ohgnarly.fileripper.enums.FileType;
import lombok.Getter;
import lombok.Setter;

import java.util.List;

@Getter @Setter
public class FileDefinition {
    private String fileMask;
    private FileType fileType;
    private boolean hasHeader;
    private String delimiter;
    private List<FieldDefinition> fieldDefinitions;
    private String recordXmlElement;
}

File: src/main/kotlin/com/ohgnarly/fileripper/models/FileDefinition.kt
package com.ohgnarly.fileripper.models


enum class FileType {
    DELIMITED,
    FIXED,
    XML
}

class FileDefinition {
    var fileMask: String = ""
    var fileType: FileType? = null
    var hasHeader: Boolean = false
    var delimiter: String = ""
    var fieldDefinitions: MutableList<FieldDefinition> = mutableListOf()
    var recordXmlElement: String = ""
    var inputDirectory: String = ""
    var completedDirectory: String = ""
}